File: apis/elysium-character-api/src/main/java/com/seventh_root/elysium/api/character/GenderProvider.java
package com.seventh_root.elysium.api.character;

import com.seventh_root.elysium.core.service.ServiceProvider;

import java.util.Collection;

public interface GenderProvider<T extends Gender> extends ServiceProvider {

    T getGender(int id);

    T getGender(String name);

    Collection<? extends T> getGenders();

    void addGender(T gender);

    void removeGender(T gender);

}

File: bukkit/elysium-characters-bukkit/src/main/kotlin/com/seventh_root/elysium/characters/bukkit/gender/GenderProvider.kt
package com.seventh_root.elysium.characters.bukkit.gender

import com.seventh_root.elysium.core.service.ServiceProvider

interface GenderProvider<T : Gender> : ServiceProvider {

    fun getGender(id: Int): T?
    fun getGender(name: String): T?
    val genders: Collection<T>
    fun addGender(gender: T)
    fun removeGender(gender: T)

}
